test/make-typed-request.js

```diff
@@ -248,3 +248,32 @@ test('request that has 500', function t(assert) {
         }
     });
 });
+
+test('request that has transformUrlFn', function t(assert) {
+    var server = createServer(function onPort(port) {
+        var treq = {
+            method: 'GET',
+            url: 'http://localhost:' + port + '/',
+            query: { prop: ['a,b', 'c,d']},
+            body: { 'hello': 'world' }
+        };
+
+        function transformUrlFn(url) {
+            // Restore all the ',' from being stringify'ed
+            return url.replace(/%2C/g, ',');
+        }
+
+        reqOpts.transformUrlFn = transformUrlFn;
```

Comment:
We are mutating the shared `reqOpts` at the top of the file.

Let's create a local `reqOpts` for this file. `var transformReqOpts = { request: request };`